claat/serve.go

```diff
@@ -0,0 +1,221 @@
+	"paper-header-panel":  "PolymerElements/paper-header-panel#^1.0.4",
+	"paper-icon-button":   "PolymerElements/paper-icon-button#^1.0.3",
+	"paper-input":         "PolymerElements/paper-input#^1.1.1",
+	"paper-item":          "PolymerElements/paper-item#^1.0.2",
+	"paper-listbox":       "PolymerElements/paper-listbox#^1.0.0",
+	"paper-menu-button":   "PolymerElements/paper-menu-button#^1.0.3",
+	"paper-dropdown-menu": "PolymerElements/paper-dropdown-menu#^1.0.4",
+	"paper-styles":        "PolymerElements/paper-styles#^1.0.11",
+	"paper-tabs":          "PolymerElements/paper-tabs#^1.1.0",
+	"paper-toast":         "PolymerElements/paper-toast#^1.0.0",
+	"paper-toolbar":       "PolymerElements/paper-toolbar#^1.0.4",
+	"platinum-sw":         "PolymerElements/platinum-sw#^1.2.0",
+	"polymer":             "Polymer/polymer#standard-layer",
+	"url-search-params":   "WebReflection/url-search-params#^0.5.0",
+}
+
+var depsDir = "bower_components"
+var elemDir = "elements"
+var elemFile = elemDir + "/codelab.html"
+
+const codelabElem = `
+<link rel="import" href="../bower_components/google-codelab-elements/google-codelab-elements.html">
+<link rel="import" href="../bower_components/google-youtube/google-youtube.html">
+<link rel="import" href="../bower_components/iron-ajax/iron-ajax.html">
+`
+
+// cmdServe is the "claat serve ..." subcommand.
+func cmdServe() {
+	err := mkdirIfNec(depsDir)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	// If we don't have the required dependencies, go get 'em.
+	for k, v := range deps {
+		if _, err := os.Stat(depsDir + "/" + k); os.IsNotExist(err) {
+			s := strings.Split(v, "#")
+			url := "https://github.com/" + s[0] + "/archive/master.zip"
+			archFile := depsDir + "/master.zip"
+			destDir := depsDir + "/" + k
+			fmt.Println("Downloading " + url)
+			err := DownloadFile(archFile, url)
+			if err != nil {
+				fmt.Println(err)
+				continue
+			}
+			err = Unzip(archFile, depsDir)
+			os.Rename(destDir+"-master", destDir)
+			os.Remove(archFile)
+			if err != nil {
+				fmt.Println(err)
+				continue
+			}
+		}
+	}
+	err = mkdirIfNec(elemDir)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	if _, err := os.Stat(elemFile); os.IsNotExist(err) {
+		f, err := os.Create(elemFile)
+		if err != nil {
+			log.Fatal("Cannot create file", err)
+		}
+		defer f.Close()
+		fmt.Fprintf(f, codelabElem)
+	}
+	fmt.Println("Dependencies installed.")
+
+	port := "9090"
+	http.Handle("/", http.FileServer(http.Dir(".")))
+	fmt.Println("Serving on localhost:" + port + ", opening browser tab now...")
+	openBrowser("http://127.0.0.1:" + port)
+	err = http.ListenAndServe(":"+port, nil) // set listen port
+	if err != nil {
+		log.Fatal("ListenAndServe: ", err)
+	}
+}
+
+// DownloadFile will download a url to a local file. It's efficient because it will
+// write as it downloads and not load the whole file into memory.
+func DownloadFile(filepath string, url string) error {
```

Comment:
Start with lower case. Don't need this function do be exported.

Reply:
fixed (though moot because this fn will be replaced soon).